src/menu/menu.ts

```diff
@@ -101,39 +102,63 @@ export class MenuUtil {
 
     return new Promise((resolve, reject) => {
 
+      if (!menuState.switches) {
+        return reject('No switches present in menu');
+      }
+
       const _quickPick = window.createQuickPick<QuickPickItem>();
 
-      _quickPick.ignoreFocusOut = true;
-      if (menuState.switches) {
-        _quickPick.items = menuState.switches.map(s => ({ label: s.shortName, detail: s.longName, description: `\t${s.description}`, activated: s.activated }));
-        _quickPick.selectedItems = _quickPick.items.filter(s => (s as any).activated);
-        _quickPick.matchOnDescription = true;
-        _quickPick.matchOnDetail = true;
-        _quickPick.canSelectMany = true;
-        _quickPick.title = 'Switches (select with <space>)';
-      }
+      _quickPick.canSelectMany = true;
+      _quickPick.title = 'Switches (type shortname of switches you want to enable)';
+      _quickPick.items = menuState.switches.map(s => ({ label: s.shortName, detail: s.longName, description: `\t${s.description}`, picked: s.activated }));
+      _quickPick.selectedItems = _quickPick.items.filter(s => s.picked);
 
-      const acceptListenerDisposable = _quickPick.onDidAccept(async () => {
+      const eventListenerDisposable = _quickPick.onDidChangeValue(async (e) => {
 
-        const updatedSwitches: Switch[] = [];
+        if (_quickPick.value === 'q') {
+          return _quickPick.hide();
+        }
+        let quickPickValue = _quickPick.value;
+        let updatedItems: QuickPickItem[] = [];
 
-        menuState.switches!.forEach(s => {
-          updatedSwitches.push({ ...s, activated: _quickPick.selectedItems.find(item => item.label === s.shortName) !== undefined });
-        });
+        for (let item of _quickPick.items) {
+
+          let match = MenuUtil.matchesSwitch(quickPickValue, item.label);
+          updatedItems.push(({ ...item, picked: match ? !item.picked : item.picked }));
```

Comment:
updatedItems is not being used?

Reply:
it was still work in progress you see. But now I think it should be good.